libraries/search/lib/objective/ObjectiveFunction.ts

```diff
@@ -97,4 +90,24 @@ export abstract class ObjectiveFunction<T extends Encoding> {
   public getLowestDistance(): number {
     return this._lowestDistance;
   }
+
+  /**
+   * Adds a child objective to the objective function
+   * @param objective the child objective
+   */
+  public addChildObjective(objective: ObjectiveFunction<T>): void {
+    this._childObjectives.push(objective);
+  }
+
+  get childObjectives(): ObjectiveFunction<T>[] {
+    return this._childObjectives;
+  }
+
+  set parentObjective(parent: ObjectiveFunction<T>) {
```

Comment:
What is the parent objective used for?

Reply:
I use it to check wether a certain objective is a "root" objective (i.e. no parents).

Reply:
Clear!